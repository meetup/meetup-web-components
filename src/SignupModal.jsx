import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

/* meetup web components */
import Modal from './interactive/Modal';
import Chunk from './layout/Chunk';
import Flex from './layout/Flex';
import FlexItem from './layout/FlexItem';
import Icon from './media/Icon';

import googleLogo from '../assets/svg/logo_google.svg';

import { C_MEDIUMGRAY } from 'swarm-constants/dist/js/constants';

/* consts */
export const SIGNUP_MODAL_CLASS = 'meetup-signupModal';
export const SIGNUP_MODAL_OR_CLASS = `${SIGNUP_MODAL_CLASS}-or`;
export const SIGNUP_MODAL_WRAPPER_CLASS = `${SIGNUP_MODAL_CLASS}-wrapper`;
export const SIGNUP_MODAL_FACEBOOK_CLASS = `${SIGNUP_MODAL_CLASS}-facebook`;
export const SIGNUP_MODAL_GOOGLE_CLASS = `${SIGNUP_MODAL_CLASS}-google`;
export const SIGNUP_MODAL_EMAIL_CLASS = `${SIGNUP_MODAL_CLASS}-email`;

/**
 * @param {Object} props component properties
 * @returns {React.element} SignupModal
 */
export const SignupModal = ({ onDismiss, signupOptions }) => {
	const { google, facebook, email, login, title, orLabel } = signupOptions;
	return (
		<Modal className={SIGNUP_MODAL_CLASS} onDismiss={onDismiss} fixed>
			<div className={SIGNUP_MODAL_WRAPPER_CLASS}>
				<Chunk>
					<h1 className="text--big">{title}</h1>
				</Chunk>
				<Chunk>
					<div className="text--secondary display--inlineBlock marginHalf--right">
						{login.text}
					</div>
					<a
						href={login.link}
						className="padding--halfLeft link display--inlineBlock"
					>
						{login.label}
					</a>
				</Chunk>
				<Chunk className="margin--top">
					<a
						href={facebook.link}
						className={cx(
							SIGNUP_MODAL_FACEBOOK_CLASS,
							'button button--fullWidth button--hasHoverShadow padding--halfLeft text--bold text--white'
						)}
					>
						<Flex>
							<FlexItem shrink className="inverted">
								<Icon
									shape="external-facebook"
									size="xs"
									className="align--left"
								/>
							</FlexItem>
							<FlexItem className="flush--left">{facebook.label}</FlexItem>
						</Flex>
					</a>
				</Chunk>
				<Chunk>
					<a
						href={google.link}
						className={cx(
							SIGNUP_MODAL_GOOGLE_CLASS,
							'button button--bordered button--fullWidth padding--halfLeft text--bold'
						)}
					>
						<Flex>
							<FlexItem shrink>
								<img src={googleLogo} className="align--left" />
							</FlexItem>
							<FlexItem className="flush--left">{google.label}</FlexItem>
						</Flex>
					</a>
				</Chunk>
				<div
					className={cx(
						SIGNUP_MODAL_OR_CLASS,
						'margin--bottom align--center text--labelSecondary text--normal'
					)}
				>
					{orLabel}
				</div>
				<Chunk>
					<a
						href={email.link}
						className={cx(
							SIGNUP_MODAL_EMAIL_CLASS,
							'button button--bordered button--fullWidth padding--halfLeft text--bold'
						)}
					>
						<Flex>
							<FlexItem shrink>
								<Icon
									shape="mail"
									size="xs"
									className="align--left"
									color={C_MEDIUMGRAY}
								/>
							</FlexItem>
							<FlexItem className="flush--left">{email.label}</FlexItem>
						</Flex>
					</a>
				</Chunk>
			</div>
		</Modal>
	);
};

SignupModal.propTypes = {
	/** Callback that happens when a user dismisses the modal */
	onDismiss: PropTypes.func,

	/** Ways a user can sign into Meetup (or sign up for Meetup) */
	signupOptions: PropTypes.object,
};

export default SignupModal;
