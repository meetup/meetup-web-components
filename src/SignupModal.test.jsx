import React from 'react';
import { shallow } from 'enzyme';

import { SignupModal, SIGNUP_MODAL_APPLE_CLASS } from './SignupModal';
import { signupOptions } from './signupModal.story';

describe('SignupModal', () => {
	const MOCK_PROPS = {
		signupOptions,
		onDismiss: jest.fn(),
	};
	const getWrapper = (props = {}) =>
		shallow(<SignupModal {...MOCK_PROPS} {...props} />);

	it('should match the snapshot', () => {
		expect(getWrapper()).toMatchSnapshot();
	});

	describe('apple signin', () => {
		describe('if apple.shouldRender is false', () => {
			it('should not show apple signin', () => {
				const props = {
					...MOCK_PROPS,
					signupOptions: {
						...MOCK_PROPS.signupOptions,
						apple: { ...MOCK_PROPS.signupOptions.apple, shouldRender: false },
					},
				};
				const wrapper = getWrapper(props);
				const appleUrl = wrapper.find(`.${SIGNUP_MODAL_APPLE_CLASS}`);
				expect(appleUrl.length).toBe(0);
			});
		});
		describe('if apple.shouldRender is true', () => {
			it('should show apple signin', () => {
				const props = {
					...MOCK_PROPS,
					signupOptions: {
						...MOCK_PROPS.signupOptions,
						apple: { ...MOCK_PROPS.signupOptions.apple, shouldRender: true },
					},
				};
				const wrapper = getWrapper(props);
				const appleUrl = wrapper.find(`.${SIGNUP_MODAL_APPLE_CLASS}`);
				expect(appleUrl.length).toBe(1);
			});
		});
	});
});
