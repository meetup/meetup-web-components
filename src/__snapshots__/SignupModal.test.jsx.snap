// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupModal should match the snapshot 1`] = `
<Modal
  className="meetup-signupModal"
  fixed={true}
  onDismiss={[MockFunction]}
>
  <div
    className="meetup-signupModal-wrapper"
  >
    <Chunk>
      <h1
        className="text--big"
      >
        Sign up
      </h1>
    </Chunk>
    <Chunk>
      <div
        className="text--secondary display--inlineBlock marginHalf--right"
      >
        Already a member?
      </div>
      <a
        className="padding--halfLeft link display--inlineBlock"
        href="meetup.com/login"
      >
        Login
      </a>
    </Chunk>
    <Chunk
      className="margin--top"
    >
      <a
        className="meetup-signupModal-facebook button button--fullWidth button--hasHoverShadow padding--halfLeft text--bold text--white"
        href="facebook.com"
      >
        <Flex>
          <FlexItem
            className="inverted"
            shrink={true}
          >
            <Icon
              className="align--left"
              shape="external-facebook"
              size="xs"
            />
          </FlexItem>
          <FlexItem
            className="flush--left"
          >
            Facebook
          </FlexItem>
        </Flex>
      </a>
    </Chunk>
    <Chunk>
      <a
        className="meetup-signupModal-google button button--bordered button--fullWidth padding--halfLeft text--bold"
        href="google.com"
      >
        <Flex>
          <FlexItem
            shrink={true}
          >
            <img
              className="align--left"
              src=""
            />
          </FlexItem>
          <FlexItem
            className="flush--left"
          >
            Google
          </FlexItem>
        </Flex>
      </a>
    </Chunk>
    <Chunk>
      <a
        className="meetup-signupModal-apple button button--bordered button--fullWidth padding--halfLeft text--bold"
        href="apple.com"
      >
        <Flex>
          <FlexItem
            shrink={true}
          >
            <img
              className="align--left"
              src=""
            />
          </FlexItem>
          <FlexItem
            className="flush--left"
          >
            Apple
          </FlexItem>
        </Flex>
      </a>
    </Chunk>
    <div
      className="meetup-signupModal-or margin--bottom align--center text--labelSecondary text--normal"
    >
      Or
    </div>
    <Chunk>
      <a
        className="meetup-signupModal-email button button--bordered button--fullWidth padding--halfLeft text--bold"
        href="meetup.com/email"
      >
        <Flex>
          <FlexItem
            shrink={true}
          >
            <Icon
              className="align--left"
              color="rgb(117, 117, 117)"
              shape="mail"
              size="xs"
            />
          </FlexItem>
          <FlexItem
            className="flush--left"
          >
            Email
          </FlexItem>
        </Flex>
      </a>
    </Chunk>
  </div>
</Modal>
`;
